internal/crawler/crawler.go

```diff
@@ -106,10 +127,49 @@ func htmlHandler(e *colly.HTMLElement) {
 	}
 }
 
-func xmlHandler(e *colly.XMLElement) {
-	err := e.Request.Visit(e.Text)
-	if err != nil && !isForbiddenURLError(err) {
-		log.Error().Err(err).Str("link", e.Text).Msg("Error attempting to visit link")
+func sitemapXmlHandler(e *colly.XMLElement) {
+	nodes, _ := xmlquery.QueryAll(e.DOM.(*xmlquery.Node), "//sitemap")
+	num_sitemaps = len(nodes)
+
+	xmlquery.FindEach(e.DOM.(*xmlquery.Node), "//sitemap", func(i int, child *xmlquery.Node) {
+		err := e.Request.Visit(child.SelectElement("loc").InnerText())
+		if err != nil && !isForbiddenURLError(err) {
+			log.Error().Err(err).Str("link", e.Text).Msg("Error attempting to visit link")
+		}
+
+		es = append(es, entry{
+			val: child.SelectElement("lastmod").InnerText(), key: child.SelectElement("loc").InnerText()})
+	})
+}
+
+func urlsetXmlHandler(e *colly.XMLElement) {
+	xmlquery.FindEach(e.DOM.(*xmlquery.Node), "//url", func(i int, child *xmlquery.Node) {
+		var lastmod string
+		if child.SelectElement("lastmod") == nil {
+			log.Info().Str("loc", child.SelectElement("loc").InnerText()).Msg("No lastmod element")
+			lastmod = "2000-01-01T00:00:00Z"
+		} else {
+			lastmod = child.SelectElement("lastmod").InnerText()
+		}
+		es = append(es, entry{
+			val: lastmod,
+			key: child.SelectElement("loc").InnerText()})
+	})
+	counter_sitemaps += 1
+}
+
+func scrapeHandler(r *colly.Response) {
+	if r.Request.URL.String() == "/sitemap.xml" || counter_sitemaps < num_sitemaps {
+		fmt.Printf("Waiting for more sitemaps to be processed: %d / %d\n", counter_sitemaps, num_sitemaps)
```

Comment:
It's better to use the logger rather than printing directly, this should probably be a log.Info().Msgf("...", counterSitemaps, numSitemaps)